### PR TITLE
Upgrade to 0.1.7.post4

### PR DESCRIPTION
**Pull request by sebp**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Bumps version to 0.1.7.post4 which adds support for numpy 2.

### CODE REVIEW

File: recipe/patches/0004-Do-not-pass-CMAKE_GENERATOR-as-variable.patch [hunk 1]
```diff
@@ -25,15 +25,15 @@ index 3790493..4f3a3a5 100644
  
  cmake_args = []
 -# What variables from the environment do we wish to pass on to cmake as variables?
--cmake_env_vars = ('CMAKE_GENERATOR', 'CMAKE_GENERATOR_PLATFORM')
+-cmake_env_vars = ('CMAKE_GENERATOR', 'CMAKE_GENERATOR_PLATFORM', 'CMAKE_OSX_ARCHITECTURES')
```
Comment (adament) [inline recipe/patches/0004-Do-not-pass-CMAKE_GENERATOR-as-variable.patch:28]: Are we sure CMAKE_OSX_ARCHITECTURES should not be passed as a definition? I am not familiar with that variable.

**Comment by adament** [comment]
Thank you very much, it looks good to me, just a small question due to my own lack of CMAKE expertise.

**Comment by sebp** [comment]
@adament What is your CMake-related question? While I did not author the original patches, maybe I can help.

**Comment by adament** [comment]
I am very sorry, I am not familiar with github reviews so I forgot to submit my question. I hope it is visible now?

File: recipe/patches/0004-Do-not-pass-CMAKE_GENERATOR-as-variable.patch (hunk 1 above)
Comment (h-vetinari) [reply recipe/patches/0004-Do-not-pass-CMAKE_GENERATOR-as-variable.patch:28]: This is the portion that the patch removes

**Review by h-vetinari — APPROVED** [review]
Thanks!

**Pull request merged**